### PR TITLE
Force update app (Fixes #419)

### LINKED ISSUE #419: check version compatibility
myPlanet will require to check version that planet supports and that is no longer supported.
then it should prompt user to upgrade application.
This will ensure that myplanet does not mesh with data in planet.
We also need to figure out what meta data myplanet will require for it and URL that planet can return to.
Related to https://github.com/open-learning-exchange/planet/issues/2688

### PR DESCRIPTION
**Pull request by rrijal53**

### CODE REVIEW

**Comment by yuviii** [comment]
@rrijal53 
Unable to update the app when working with my local server
To test i did
 lowered the version in build.gradle
 ```
  versionCode 227
  versionName "0.2.27"
```

**Server response :**  Failed to connect to /192.168.100.224:80

Mine server is running in 2400 port. I think we should add the port if it is present in url for version as well. Currently it is adding default http request port 80 itself.

Myplanet setting configuration


screenshot_1545843981

**Comment by rrijal53** [comment]
@yuviii  it only works on 80 port. All raspberry pi should uses 80 port for planet for now . To address your problem we need to ask user for planet url and database url which will be more confusing.

This problem will not arise when we solve https://github.com/open-learning-exchange/myplanet/issues/414 this issue.

**Comment by yuviii** [comment]
@rrijal53 great, but app is allowing me to login though the version check is failed. I think we should stop app from been login if version check is failed as update might be compulsory to avoid crash.

**Comment by lmmrssa** [comment]
@yuviii if myplanet is not able to connect to planet then it should not stop app from running. MyPlanet is supposed to work offline.

**Comment by lmmrssa** [comment]
Right now version check is only done on app load.
* When there is no url versions will not be available so it will not start sync
* When there is url versions is first thing it should check before calling sync
* If versions is not found or is not valid then syncing should not start
* If versions is found and valid then only sync should start
* Version check should always be called before any sync done. Manual or Auto-sync

File: app/build.gradle [hunk 1]
```diff
@@ -7,7 +7,7 @@ android {
         applicationId "org.ole.planet.myplanet"
         minSdkVersion 19
         targetSdkVersion 27
-        versionCode 229
+        versionCode 228
```
Comment (lmmrssa) [inline app/build.gradle]: fix this

Comment (rrijal53) [reply app/build.gradle]: Fixed :)

**Pull request merged**